### PR TITLE
Wave Sort in C++

### PR DESCRIPTION
**Pull request by gayatri517**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Code added in the sorting folder.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2228

### CODE REVIEW

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    return 0;
+}
+//Example 1:
+//Input[] : {20 80 40 35 10 15 70}
+//Output[] : {80 20 40 10 35 15 70}
+
+//Example 2:
+//Input[] = {12 20 45 60 5}
+//Output[] : {20 12 60 5 45}
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Put it in Multi-Line Comments and add Complexity

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        cin>>temp;
+        v.push_back(temp);
+    }
+    wave_sort(v,n); 
+    //Printing the final vector
+    for(int i=0;i<n;i++) {
+        cout<<v[i]<<" ";
+    }
+    return 0;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Format and Indent the Code properly

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
+//Problem statement :
+//Given an array of integers, sort the array into a wave like arrangement,
+//that is A[0 to n-1] is sorted in the form A[0]>=A[1]<=A[2]>=A[3]....
+
+#include<bits/stdc++.h>
+using namespace std;
+vector<int> wave_sort(vector<int> &v,int n) {
+    for(int i=0;i<n;i+=2) {
+        if(v[i]<v[i-1] and i!=0) { //If the current element is lesser in value than the previous and is not the first..
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Put Comments above the line of code, not beside them

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            swap(v[i],v[i-1]); //then the values are swapped
+        }
+        if(v[i]<v[i+1] and i!=n-1) { //If the current element is lesser in value than the next and is not the last..
+            swap(v[i],v[i+1]); //then the values are swapped
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Put Comments above the line of code, not beside them

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
+//Problem statement :
+//Given an array of integers, sort the array into a wave like arrangement,
+//that is A[0 to n-1] is sorted in the form A[0]>=A[1]<=A[2]>=A[3]....
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Add more descriptive Problem Statement

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by gayatri517** [comment]
@HarshCasper I have made the changes that were requested. Please check them. 
Thank you.

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+    //Printing the final vector
+
+    for(int i=0;i<n;i++) {
+    	
+        	cout<<v[i]<<" ";
+    }
+
+    return 0;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Why do we have so many lines of code missed out?

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    	/* If the current element is lesser in value than the previous and is not the first.
+    	then the values are swapped */
+        if(v[i]<v[i-1] and i!=0) { 
+            swap(v[i],v[i-1]); 
+        }
+
+        /* If the current element is lesser in value than the next and is not the last.
+        then the values are swapped */
+        if(v[i]<v[i+1] and i!=n-1) { 
+            swap(v[i],v[i+1]);
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Format the Code

File: C-Plus-Plus/sort/wave_sort.cpp [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        	v.push_back(temp);
+    }
+
+    wave_sort(v,n); 
+
+    //Printing the final vector
+
+    for(int i=0;i<n;i++) {
+    	
+        	cout<<v[i]<<" ";
```
Comment (HarshCasper) [inline C-Plus-Plus/sort/wave_sort.cpp]: Format the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by gayatri517** [comment]
@HarshCasper formatting done. Please check them.
Thank you.

**Review by HarshCasper — APPROVED** [review]

**Comment by gayatri517** [comment]
@HarshCasper this PR is not getting included in the score in leaderboard. Please check it.
Thank you

**Pull request merged**